jax/scipy/stats/uniform.py

```diff
@@ -20,14 +20,15 @@
 import scipy.stats as osp_stats
 
 from ... import lax
-from ...numpy.lax_numpy import _promote_args_like, _wraps, where, inf, logical_or
+from ...numpy.lax_numpy import _promote_args_like, _wraps, where, inf, logical_or, broadcast_to
 
 
 @_wraps(osp_stats.uniform.logpdf)
 def logpdf(x, loc=0, scale=1):
   x, loc, scale = _promote_args_like(osp_stats.uniform.logpdf, x, loc, scale)
   fill_value = lax.neg(lax.log(scale))
-  log_probs = lax.broadcast(fill_value, onp.shape(x))
```

Comment:
Sorry for not noticing this before, but I think there's a simpler fix.

Just set `log_probs = lax.neg(lax.log(scale))`, and don't have any additional broadcasting here.

The reason this should work is because `_promote_args_like` also promotes the shape by making all of the arguments of equal rank. And all the other operators here broadcast too. So we shouldn't need any additional broadcasting.

Reply:
Nice observation!